aws/resource_aws_quicksight_user.go

```diff
@@ -0,0 +1,232 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/quicksight"
+)
+
+func resourceAwsQuickSightUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsQuickSightUserCreate,
+		Read:   resourceAwsQuickSightUserRead,
+		Update: resourceAwsQuickSightUserUpdate,
+		Delete: resourceAwsQuickSightUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"aws_account_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"email": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"iam_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"identity_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"IAM",
+					"QUICKSIGHT",
+				}, false),
+			},
+
+			"namespace": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "default",
+				ValidateFunc: validation.StringInSlice([]string{
+					"default",
+				}, false),
+			},
+
+			"session_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"user_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"user_role": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"READER",
```

Comment:
Same here regarding constants:

```suggestion
					quicksight.UserRoleReader,
```

It also appears (at least in the constants) that there are two additional values available:

```
    // UserRoleRestrictedAuthor is a UserRole enum value
    UserRoleRestrictedAuthor = "RESTRICTED_AUTHOR"

    // UserRoleRestrictedReader is a UserRole enum value
    UserRoleRestrictedReader = "RESTRICTED_READER"
```

Reply:
@bflad I use QuickSight a good deal in my day job and noticed these as well a few months ago. I reached out to Amazon and they notified me that these are internal and should not be used, haha.

Reply:
Haha nice 😂 Thanks!